### PR TITLE
NEW Add update extension hooks for LinkFormFactory subclasses

### PR DESCRIPTION
**Pull request by robbieaverill**
Requires https://github.com/silverstripe/silverstripe-admin/pull/661

Then closes https://github.com/silverstripe/silverstripe-cms/issues/2273

### CODE REVIEW

**Comment by unclecheese** [comment]
Have retargeted toward `4.3`.

File: code/Forms/InternalLinkFormFactory.php [hunk 1]
```diff
@@ -48,6 +48,8 @@ protected function getFormFields($controller, $name, $context)
             $fields->insertAfter('PageID', TextField::create('Text', _t(__CLASS__.'.LINKTEXT', 'Link text')));
         }
 
+        $this->extend('updateFormFields', $actions, $controller, $name, $context);
```
Comment (ScopeyNZ) [inline code/Forms/InternalLinkFormFactory.php]: `$fields`?

Comment (robbieaverill) [reply code/Forms/InternalLinkFormFactory.php]: Copy and paste fail- busted

**Comment by robbieaverill** [comment]
Last one to close the issue here @ScopeyNZ

**Pull request merged**